library/src/main/java/com/schibsted/spain/barista/internal/MagicAssert.kt

```diff
@@ -0,0 +1,63 @@
+package com.schibsted.spain.barista.internal
+
+import android.support.test.espresso.Espresso.onView
+import android.support.test.espresso.assertion.ViewAssertions
+import android.support.test.espresso.matcher.ViewMatchers.withText
+import android.view.View
+import com.schibsted.spain.barista.internal.failurehandler.SpyFailureHandler
+import com.schibsted.spain.barista.internal.failurehandler.description
+import com.schibsted.spain.barista.internal.matcher.HelperMatchers.firstViewOf
+import com.schibsted.spain.barista.internal.util.resourceMatcher
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.Matchers.not
+
+/**
+ * Extension function alias for [magicAssertOnView] supporting both Id or String resources.
+ */
+infix fun Int.magicAssert(condition: Matcher<View>) {
+    this.resourceMatcher() magicAssert condition
+}
+
+/**
+ * Extension function alias for [magicAssertOnView] using [withText] as viewMatcher
+ */
+infix fun String.magicAssert(condition: Matcher<View>) {
+    withText(this) magicAssert condition
+}
+
+/**
+ * Extension function alias for [magicAssertOnView]
+ */
+infix fun Matcher<View>.magicAssert(condition: Matcher<View>) {
+    magicAssertOnView(viewMatcher = this, condition = condition)
+}
+
+/**
+ * Performs an assertion of a [condition] on a view described by [viewMatcher].
+ *
+ * Attempts to assert using multiple scenarios for the [viewMatcher]:
+ * 1. Just one view matches the [viewMatcher].
+ * 2. Multiple views match the [viewMatcher]: will pass if at least one of them matches the [condition].
+ */
+fun magicAssertOnView(viewMatcher: Matcher<View>, condition: Matcher<View>) {
+    val spyFailureHandler = SpyFailureHandler()
+    try {
+        onView(viewMatcher)
+                .withFailureHandler(spyFailureHandler)
+                .check(ViewAssertions.matches(condition))
+    } catch (firstError: RuntimeException) {
```

Comment:
@SmasSive look! A try/catch in the wild!

I tried implementing this with some kind of [Try monad](http://kategory.io/docs/datatypes/try/) to avoid anidating these try/catch, and it worked. But I'm not sure it's worth that complexity just for one method 😕  (or two in the future).

Reply:
Here's what I was able to implement with Kategory.
<img width="1068" alt="captura de pantalla 2017-12-18 a las 9 17 20" src="https://user-images.githubusercontent.com/545251/34096092-555137c6-e3d4-11e7-8b88-a573e1a36a6c.png">

It might be simplified even more with some custom structure instead of this one. Not sure if that's enough though. 
Thoughts? 💭

Reply:
I do love the implementation with Kategory, it simplifies a lot the code. I'd love it even more if you separate the `Try`s in methods, just to improve readability, something like:
```kotlin
tryToAssertView()
  .recoverWith { tryToAssertFirstViewOf() }
  .getOrElse { spy... }
```
What do you think?

Reply:
You know I hate you, right? 🤣 
Ok let me see what I can do.

Reply:
Here are a couple of examples I've put together.

With Kategory:
<img width="1058" alt="captura de pantalla 2017-12-19 a las 10 55 19" src="https://user-images.githubusercontent.com/545251/34156531-eba052f4-e4bd-11e7-9597-eb5aa08cbf78.png">
With a custom DSL-ish:
<img width="1121" alt="captura de pantalla 2017-12-19 a las 12 03 55" src="https://user-images.githubusercontent.com/545251/34156549-fedd7914-e4bd-11e7-8628-0bb68ee9c46c.png">

I like how the second one looks. But again, I'm not sure if we're being pragmatic here. Seems like a lot of complexity for something that will be use in just two places.

Reply:
I like the second option too but the key here I think is to see how it is implemented, if it adds a lot of complexity I'd discard it but if it doesn't, go for it!
BTW, I invoke some of the usual contributors like @rocboronat @alorma @sergiandreplace to light us up here :dancing_men:

Reply:
Well... For the implementation I could use a hand.

This is something that works, not too obvious though:
<img width="639" alt="captura de pantalla 2017-12-19 a las 14 45 49" src="https://user-images.githubusercontent.com/545251/34160127-f8d9c7f8-e4cb-11e7-8bba-91404adf1c85.png">

Reply:
I'm starting to think the initial double try/catch is the best solution 🤣

Reply:
Ok, looks complex... So my call would go for the first implementation with Kategory.
Please please please, don't kill me ❤️